tinywl/tinywl.c

```diff
@@ -709,7 +705,9 @@ static void xdg_toplevel_map(struct wl_listener *listener, void *data) {
 
 	wl_list_insert(&toplevel->server->toplevels, &toplevel->link);
 
-	focus_toplevel(toplevel, toplevel->xdg_toplevel->base->surface);
+	struct wlr_surface *surface = toplevel->xdg_toplevel->base->surface;
+	wlr_scene_shadow_set_size(toplevel->shadow, surface->current.width, surface->current.height); // TODO: use get node size?
```

Comment:
This should be the xdg geometry. The surface size would include potential csd dimensions

Reply:
Yup, left a comment about that. I believe the get_node_size function does all that for us, plus works with rects as well